src/main/java/com/echobox/api/linkedin/client/DefaultWebRequestor.java

```diff
@@ -574,10 +576,24 @@ protected Response fetchResponse(HttpResponse httpUrlConnection) throws IOExcept
    */
   protected Response fetchResponse(int statusCode, HttpHeaders headers, String body) {
     Map<String, String> headerMap = headers.entrySet().stream()
-        .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().toString(),
-            (key, value) -> value));
+            .collect(Collectors.toMap(Map.Entry::getKey, headerValueMapper()));
     return new Response(statusCode, headerMap, body);
   }
+  
+  private Function<Map.Entry<String, Object>, String> headerValueMapper() {
+    return entry -> {
+      List<Object> value = (List<Object>) entry.getValue();
```

Comment:
Would it be better to check if `entry.getValue()` is `List `before casting?
If it is not a `List`, we can just return `toString()`

Reply:
From what I've seen, each header value is contained in a list. I've added a check before casting for added protection